### DISCUSSION TITLE
Use `DATABASE_URL_TEST` when runnig `prisma migrate`

### DISCUSSION THREAD

**Discussion by joaocasarin** [prisma/prisma#15477] [Q&A] [answered]
Well, basically I NEED to use a second database url when executing my migrations for test purposes, and with that I want to avoid having multiple `.env` files and using the `dotenv-cli` to force both jest and `npx prisma migrate` to use a custom `.env.test` with the correct value in the `DATABASE_URL`.

Can we do that?

**Comment by janpio** [comment, marked as answer]
You could adapt this pattern: https://www.prisma.io/docs/data-platform/data-proxy#providing-the-connection-string-to-prisma-migrate

**Reply by joaocasarin** [reply]
wow, that actually seems great!
Can I assume that the variable `MIGRATE_DATABASE_URL` will be read from the `.env` file in the PWD? Also, I am not sure if that would work on Windows, any info regarding that?

**Reply by janpio** [reply]
It will read the variable from the environment, and using `.env` is one way to get a value from a file into the environment, yes.
On Windows you would need to use https://www.npmjs.com/package/cross-env

**Reply by joaocasarin** [reply]
I have tried with `"pretest": "cross-env DATABASE_URL=\"${DATABASE_URL_TEST}\" npx prisma migrate deploy"` and `"pretest": "cross-env DATABASE_URL=${DATABASE_URL_TEST} npx prisma migrate deploy"` and it did not work... the only way I could make it work was using `dotenv-cli` as well, becoming this script: `"pretest": "dotenv -e .env cross-env DATABASE_URL=${DATABASE_URL_TEST} npx prisma migrate deploy"`.

Any suggestion to avoid that concatenation of 2 packages with the prisma deploy command?

Oh, and I have noticed that I need to use that `DATABASE_URL=${DATABASE_URL_TEST}` when running jest as well, thus having these commands, is that right?

```json
"pretest": "dotenv -e .env cross-env DATABASE_URL=${DATABASE_URL_TEST} npx prisma migrate deploy",
"test": "dotenv -e .env cross-env DATABASE_URL=${DATABASE_URL_TEST} jest --no-cache -i",
```

**Reply by janpio** [reply]
You are right, if you want to modify the env vars before the `prisma` CLI is actually active, you indeed also need to use `dotenv` manually. The example I linked to above assumes the variable to already be in the actual environment - not a file.

**Reply by joaocasarin** [reply]
> You are right, if you want to modify the env vars before the `prisma` CLI is actually active, you indeed also need to use `dotenv` manually. The example I linked to above assumes the variable to already be in the actual environment - not a file.

okay, that's not a problem at all. Your solution indeed worked, I'd just like to know if there's any plan on doing such things without using packages like cross-env and dotenv-cli... Maybe a conditional database url in the `prisma.schema`?

**Reply by janpio** [reply]
There is an issue for this somewhere, but it makes a _lot_ of things a lot more complicated (just debugging where the connection string comes from that might not be working) so right now we are not super hot as long as we can get good workarounds.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
